stable/wordpress/templates/deployment.yaml

```diff
@@ -28,21 +28,47 @@ spec:
         - name: MARIADB_ROOT_PASSWORD
           valueFrom:
             secretKeyRef:
+            {{- if .Values.mariadb.enabled }}
               name: {{ template "mariadb.fullname" . }}
               key: mariadb-root-password
+            {{- else }}
+              name: {{ printf "%s-%s" .Release.Name "externaldb" }}
+              key: db-root-password
+            {{- end }}
         - name: MARIADB_HOST
+        {{- if .Values.mariadb.enabled }}
           value: {{ template "mariadb.fullname" . }}
+        {{- else }}
+          value: {{ default "localhost" .Values.externalDatabase.hostname | quote }}
+        {{- end }}
         - name: MARIADB_PORT_NUMBER
+        {{- if .Values.mariadb.enabled }}
           value: "3306"
+        {{- else }}
+          value: {{ default "3306" .Values.externalDatabase.port | quote }}
+        {{- end }}
         - name: WORDPRESS_DATABASE_NAME
+        {{- if .Values.mariadb.enabled }}
           value: {{ default "" .Values.mariadb.mariadbDatabase | quote }}
+        {{- else }}
+          value: {{ default "bitnami_wordpress" .Values.externalDatabase.dbName | quote }}
```

Comment:
can we call this externalDatabase.database as in Ghost?

Reply:
The defaults shouldn't be defined here, they should be uncommented in the values.yaml. We used to use `{{ default ...` in charts but are moving away from it in favour of clear defaults in values.yaml.

Reply:
It is also in values.yaml (but commented out)
"The defaults shouldn't be defined here"  My interpretation of this statement is going down a bad user experience route, can you clarify a little (or point to appropriate guidance) ?
Are you saying those values should be uncommented?
Is it not going to cause confusion with some people having optional values uncommented ? 
Wouldn't people then be asking "Why do I need to supply these values if I'm using the standard embedded mariadb?"
If we leave the defaults commented out, then if someone sets mariadb.enabled to false we will be getting empty strings and so bad deploys.
eg
`
        - name: MARIADB_HOST
          value: 
`

Reply:
Sorry for not being clear. I am indeed saying that the values should be uncommented. This allows people to see the defaults without having to go through the templates. Additionally, it allows for the defaults in values.yaml to be machine readable (and therefore parseable by something like Monocular, for example).

> Wouldn't people then be asking "Why do I need to supply these values if I'm using the standard embedded mariadb?"

A fair point, but something can be cleared up by a comment explaining that the values are only used if `mariadb.enabled` is false.